packages/flutter/lib/src/services/system_chrome.dart

```diff
@@ -228,6 +231,24 @@ List<String> _stringify(List<dynamic> list) {
 class SystemChrome {
   SystemChrome._();
 
+  /// Whether Flutter should automatically compute the desired system UI.
+  ///
+  /// When this setting is enabled, Flutter will hit-test the layer tree at the
+  /// top and bottom of the screen on each frame looking for an
+  /// [AnnotatedRegionLayer] with an instance of a [SystemUiOverlayStyle]. The
+  /// hit-test result from the top of the screen provides the status bar settings
+  /// and the hit-test result from the bottom of the screen provides the system
+  /// nav bar settings.
+  ///
+  /// If you want to imperatively set the system ui style instead, it is
+  /// recommended that [automaticSystemUiAdjustment] is set to false.
+  ///
+  /// See also:
+  ///
+  ///   * [AnnotatedRegion], for placing [SystemUiOverlayStyle] in the layer tree.
+  ///   * [SystemChrome.setSystemUIOverlayStyle], for imperatively setting the system ui style.
+  static bool automaticSystemUiAdjustment = true;
```

Comment:
This should probably be a feature of RenderView rather than a global static, because otherwise if you have two trees (e.g. one on-screen and one off-screen, or both on-screen but one being embedded in the other), you can't turn it off for one tree but leave it on for the "real" tree.

Reply:
Moved